libr/core/cmd.c

```diff
@@ -1266,6 +1282,239 @@ static int cmd_stdin(void *data, const char *input) {
+				separator = "|";
+				continue;
+			}
+			if (r_str_startswith (line, "│─")) {
+				expect_rows = true;
+				separator = "│";
+				continue;
+			}
+		}
+
+		RTableColumnType *typeString = r_table_type ("string");
+		RTableColumnType *typeNumber = r_table_type ("number");
+		if (expect_header) {
+			char *arg;
+			RList *args = r_str_split_list (line + strlen (separator), separator, 0);
+			RListIter *iter2;
+			ncols = 0;
+			if (r_list_length (t->cols) > 0) {
+				eprintf ("Warning: Not re-adding headers. Use ,- to reset the table.\n");
+				continue;
+			}
+			r_list_foreach (args, iter2, arg) {
+				char *s = strchr (arg, ' ');
+				char *ss = r_str_trim_dup (s? s + 1: arg);
+				if (!*ss) {
+					free (ss);
+					continue;
+				}
+				r_table_add_column (t, typeString, ss, 0);
+				ncols ++;
+			}
+			expect_header = false;
+		} else if (expect_rows) {
+			char *arg;
+			RList *args = r_str_split_list (line + strlen (separator), separator, 0);
+			RList *items = r_list_newf (free);
+			RListIter *iter2;
+			if (r_list_length (args) < ncols) {
+				// dowarn?
+				continue;
+			}
+			r_list_foreach (args, iter2, arg) {
+				char *ss = r_str_trim_dup (arg);
+				if (!*ss) {
+					free (ss);
+					continue;
+				}
+				if (isdigit (*ss)) {
+					int col = r_list_length (items);
+					RTableColumn *c = r_list_get_n (t->cols, col);
+					if (c) {
+						c->type = typeNumber;
+					}
+				}
+				r_list_append (items, ss);
+			}
+			RTableRow *row = r_table_row_new (items);
+			r_list_append (t->rows, row);
+		}
+	}
+	r_list_free (lines);
+	free (data);
+}
+
+static int cmd_table(void *data, const char *input) {
+	RCore *core = (RCore*)data;
+
+	if (!core->table) {
+		core->table = r_table_new ();
+	}
+
+	switch (*input) {
+	case 'h': // table header
+		{
+			char *s = r_str_trim_dup (input + 1);
+			RList *list = r_str_split_list (s, " ", 0);
+			RListIter *iter;
+			char *format = r_list_pop_head (list);
+			if (!format) {
+				break;
+			}
+			size_t i = 0;
+			r_list_foreach (list, iter, s) {
+				const char type_char = format[i];
+				if (!type_char) {
+					break;
+				}
+				const char *type_name = (type_char == 's')
+					? "string": "number";
+				RTableColumnType *typeString = r_table_type (type_name);
+				r_table_add_column (core->table, typeString, s, 0);
+				i++;
+			}
+			r_list_free (list);
+			free (format);
+	//		free (s);
+		}
+		break;
+	case 'r': // add row
+		{
+			char *args = r_str_trim_dup (input + 1);
+			RList *list = r_str_split_list (args, " ", 0);
+			if (list) {
+				r_table_add_row_list (core->table, list);
+			}
+			// args is now owned by the list
+			//free (args);
+		}
+		break;
+	case '-':
+		r_table_free (core->table);
+		core->table = r_table_new ();
+		break;
+	case '/':
+		// query here
+		if (r_table_query (core->table, input + 1)) {
+			char *ts = r_table_tostring (core->table);
+			r_cons_printf ("%s", ts);
+			free (ts);
+		}
+		break;
+	case '$':
+		load_table (core, core->table, input);
```

Comment:
get the `data` from the alias before calling `load_table` and call it just with the data, so `load_table` doesn't need to perform the loading from files and it can potentially get data from everywhere, not just aliases and files.

Reply:
done